File: frontend/src/api/agent.ts
import { api } from "./http";

export type AgentAction = {
  id: string;
  profil_id: string;
  type_action: string;
  payload_json: string;
  statut: string;
  message?: string | null;
  created_at: string;
};

export type AgentDigest = {
  alertes_stock: Array<Record<string, unknown>>;
  budget: Record<string, unknown>;
  ce_soir: Record<string, unknown> | null;
  actions: AgentAction[];
  memories: Array<Record<string, unknown>>;
  resume: string;
};

export function getAgentDigest(profilId: string, token: string) {
  return api<AgentDigest>(`/ia/${profilId}/agent/digest`, { token });
}

export function respondAgentAction(
  profilId: string,
  token: string,
  actionId: string,
  decision: "accepte" | "refuse"
) {
  return api<AgentAction>(
    `/ia/${profilId}/agent/actions/${actionId}/respond`,
    { method: "POST", token, body: { decision } }
  );
}

File: backend/schemas/agent.py
from datetime import datetime
from typing import Any, Literal, Optional

from pydantic import BaseModel, ConfigDict, Field


class AgentActionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    profil_id: str
    type_action: str
    payload_json: str
    statut: str
    message: Optional[str] = None
    created_at: datetime


class AgentDigestOut(BaseModel):
    alertes_stock: list[dict[str, Any]] = Field(default_factory=list)
    budget: dict[str, Any] = Field(default_factory=dict)
    ce_soir: Optional[dict[str, Any]] = None
    actions: list[AgentActionOut] = Field(default_factory=list)
    memories: list[dict[str, Any]] = Field(default_factory=list)
    resume: str = ""


class AgentActionRespondRequest(BaseModel):
    decision: Literal["accepte", "refuse"]
